controller/hardware/actuator/board-neoled.cpp

```diff
@@ -0,0 +1,164 @@
+#include <actuator/board-neoled.hpp>
+#include <bitset>
+#include <cstdlib>
+#include <cstring>
+#include <limits>
+#include <stm-hal/hal-tim.hpp>
+#include <stm-hal/hal-gpio.hpp>
+#include <gscope/gscope.hpp>
+#include <gscope/gscope.hpp>
+
+// IN-PI554FCH - https://www.inolux-corp.com/datasheet/SMDLED/Addressable%20LED/IN-PI554FCH.pdf
+
+// ------------------------------------------------------------------------------------
+//                                      SYSTEM API
+// ------------------------------------------------------------------------------------
+
+static bool s_sending_data = false;
+
+static void s_timer_transfer_finish_callback(void*)
+{
+    // after this time, timer is ready to send a new color
+    s_sending_data = false;
+}
+
+// ------------------------------------------------------------------------------------
+//                                      PUBLIC API
+// ------------------------------------------------------------------------------------
+
+/**
+ * Init Neoled.
+ *
+ * Calls system_neoled_init()
+*/
+void Neoled::init()
+{
+    uint32_t timer_cnt_frequency = hal_tim_neoled_init(s_timer_transfer_finish_callback,  nullptr);
+    m_rgb_timer_on_off_periods[0] = timer_cnt_frequency * 0.0000003f;   // 0
+    m_rgb_timer_on_off_periods[1] = timer_cnt_frequency * 0.0000006f;   // 1
+}
+
+/**
+ * Update Neoled.
+ *
+ * If the colour has changed then it will load the new RGB values into the LED.
+*/
+void Neoled::update()
+{
+    uint32_t now = hal_timer_32_ms();
+    static uint32_t s_timestamp;
```

Comment:
this shouldn't be static, and instead, it should be declared private to the class